src/smart-components/workspaces/workspaces.tsx

```diff
@@ -1,11 +1,71 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { useIntl } from 'react-intl';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchWorkspaces } from '../../redux/actions/workspaces-actions';
 import messages from '../../Messages';
 import { ContentHeader } from '@patternfly/react-component-groups';
-import { Card, PageSection } from '@patternfly/react-core';
+import { PageSection } from '@patternfly/react-core';
+import { DataView, DataViewTable, DataViewTh, DataViewTrTree, useDataViewSelection } from '@patternfly/react-data-view';
+import { Workspace } from '../../redux/reducers/workspaces-reducer';
+import { RBACStore } from '../../redux/store';
 
 const Workspaces = () => {
   const intl = useIntl();
+  const dispatch = useDispatch();
+  const selection = useDataViewSelection({ matchOption: (a, b) => a.id === b.id });
+
+  const { isLoading, workspaces, error } = useSelector((state: RBACStore) => state.workspacesReducer);
+
+  const [hierarchicalWorkspaces, setHierarchicalWorkspaces] = useState<Workspace[]>([]);
+
+  useEffect(() => {
+    dispatch(fetchWorkspaces());
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (workspaces.length > 0) {
+      const mapWorkspacesToHierarchy = (workspaceData: any[]): Workspace[] => {
+        const workspaceMap: { [key: string]: Workspace } = {};
+
+        workspaceData.forEach((ws) => {
+          workspaceMap[ws.uuid] = {
+            id: ws.uuid,
+            name: ws.name,
+            description: ws.description,
+            children: [],
+          };
+        });
+
+        const hierarchy: Workspace[] = [];
+        workspaceData.forEach((ws) => {
+          if (ws.parent_id) {
+            workspaceMap[ws.parent_id]?.children?.push(workspaceMap[ws.uuid]);
+          } else {
+            hierarchy.push(workspaceMap[ws.uuid]);
+          }
+        });
+
+        return hierarchy;
+      };
```

Comment:
```suggestion
      const mapWorkspacesToHierarchy = (workspaceData: any[]): Workspace[] => {
        const workspaceMap: { [key: string]: Workspace } = {};

        workspaceData.forEach((ws) => {
          workspaceMap[ws.uuid] = {
            id: ws.uuid,
            name: ws.name,
            description: ws.description,
            children: [],
          };
        });

        const hierarchy: Workspace[] = [];
        workspaceData.forEach((ws) => {
          if (ws.parent_id) {
            workspaceMap[ws.parent_id]?.children?.push(workspaceMap[ws.uuid]);
          } else {
            hierarchy.push(workspaceMap[ws.uuid]);
          }
        });

        return hierarchy;
      };
```

Maybe we could move this above the component just for the readability

Reply:
Function declaration moved out of useEffect hook.